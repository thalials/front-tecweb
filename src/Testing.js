import React, { useEffect, useState, useLayoutEffect } from "react";

export default function Testing() {
  return (
    <div>
      <p>oi</p>
    </div>
  );
}
