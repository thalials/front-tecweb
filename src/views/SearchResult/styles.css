.MuiCardActions-root{
  /* background-color: aqua;  */
}

.search-result-outer {
  display: flex;
  justify-content: center;
}

.search-result-card {
  background-color: #f6f6f6;
  padding: 1rem;
  box-sizing: border-box;
  width: 80vw;
  max-width: 800px;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
}

.result-card-outer {
  display: grid;
  row-gap: 0.5rem;
}

.city-map {
  /* display: flex; */
  max-width: 800px;
  max-height: 300px;
  box-shadow: 1px 1px 20px rgba(0, 0, 0, 0.13);
}

.search-result-main {
  height: 90vh;
  justify-content: center;
  display: block;
  margin: 0 0 0 0;
}

.result-card-outer * {
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-weight: 400;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.result-card-outer ul {
  margin: 0.5rem 0;
}

.result-card-outer ul li {
  margin: 0.5rem;
}

.print-share {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  /* background-color: blue; */
}
.wiki {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  /* background-color: blueviolet; */
  width: 100%;
}
.insta_logo {
  width: 100px;
  height: 100px;
}

.share-qrcode {
  padding: 0.5rem;
  background-color: white;
  box-shadow: 0px 1px 20px #ccc;
}

.label {
  font-weight: bold;
}

.likes {
    row-gap: 0;
}

.dislikes {
    row-gap: 0;
}