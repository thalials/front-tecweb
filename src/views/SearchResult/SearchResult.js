import React, { useEffect, useState, useLayoutEffect } from "react";
import { Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import clsx from "clsx";
import Card from "@material-ui/core/Card";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import { CardHeader } from "@material-ui/core";
import CardActions from "@material-ui/core/CardActions";
import Collapse from "@material-ui/core/Collapse";
import Avatar from "@material-ui/core/Avatar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import { red } from "@material-ui/core/colors";
import {
  Favorite,
  FavoriteBorderOutlined,
  ThumbUp,
  ThumbUpOutlined,
  ThumbDown,
  ThumbDownAltOutlined,
} from "@material-ui/icons";
import ShareIcon from "@material-ui/icons/Share";
import ExpandMoreIcon from "@material-ui/icons/ExpandMore";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import { ReactComponent as ReactLogo } from "../../assets/logo_wiki.svg";
import { ReactComponent as ReactLogo1 } from "../../assets/instagram.svg";
import { ReactComponent as ReactLogo2 } from "../../assets/facebook.svg";
import { ReactComponent as ReactLogo3 } from "../../assets/panorama.svg";
import { ReactComponent as ReactLogo4 } from "../../assets/youtube.svg";
import { ReactComponent as ReactLogoMap } from "../../assets/info.svg";

import {
  createQRCodeURI,
  toggleLike,
  getCityInfo,
  toggleDislike,
} from "../../API/Requests";
import { getCityId } from "../../Helpers";
import "./styles.css";
import Header from "../../Components/Header";
import LoadingIndicator from "../../Components/LoadingIndicator/LoadingIndicator";

function SearchResult(props) {
  const _id = props.match.params.city_id;

  return (
    <div>
      <Header />
      <section className="search-result-main">
        <ResultCard id={_id} />
      </section>
    </div>
  );
}

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 345,
  },
  media: {
    height: 0,
    paddingTop: "56.25%", // 16:9
  },
  expand: {
    transform: "rotate(0deg)",
    marginLeft: "auto",
    transition: theme.transitions.create("transform", {
      duration: theme.transitions.duration.shortest,
    }),
  },
  expandOpen: {
    transform: "rotate(180deg)",
  },
  avatar: {
    backgroundColor: red[500],
  },
}));

function ResultCard(props) {
  const { id } = props;
  const [city, setCity] = useState({});
  const [country, setCountry] = useState({});
  const [expanded, setExpanded] = useState(true);
  const [loading, setLoading] = useState(true);
  const [likedByMe, setLikedByMe] = useState(false);
  const [likes, setLikes] = useState(0);
  const [dislikedByMe, setDislikedByMe] = useState(false);
  const [dislikes, setDislikes] = useState(0);
  const [error, setError] = useState(false);
  const [cityName, setCityName] = useState();

  const qrCodeURI = createQRCodeURI(window.location.href);

  const splittedCityName = city?.name?.split(/\s/);
  const acronym = splittedCityName?.reduce(
    (response, word) => (response += word.slice(0, 1)),
    ""
  );

  const classes = useStyles();

  const handleExpandClick = () => {
    setExpanded(!expanded);
  };

  async function handleLike() {
    toggleLike(id).then((placeIsLiked) => {
      setLikedByMe(placeIsLiked);
      if (placeIsLiked) {
        setLikes((prev) => prev + 1);
      } else {
        setLikes((prev) => prev - 1);
      }
    });
  }

  async function handleDislike() {
    toggleDislike(id).then((placeIsDisliked) => {
      setDislikedByMe(placeIsDisliked);
      if (placeIsDisliked) {
        setDislikes((prev) => prev + 1);
      } else {
        setDislikes((prev) => prev - 1);
      }
    });
  }

  const curTime = new Date().toLocaleString();

  useLayoutEffect(() => {
    setLoading(true);
    (async () => {
      await getCityInfo(id)
        .then((city) => {
          setCountry(city.country);
          setCity(city);
          setLikes(city.likes);
          setLikedByMe(city.likedByMe);
          setDislikes(city.dislikes);
          setDislikedByMe(city.dislikedByMe);
          setCityName(city.name);
        })
        .catch(() => {
          setError(true);
        });
    })().then(() => {
      setLoading(false);
    });
  }, [id]);

  return (
    <>
      {error ? (
        <EmptyFeedback />
      ) : loading ? (
        <LoadingIndicator width={30} />
      ) : (
        <div
          className="search-result-outer"
          style={{
            background: `url(https://source.unsplash.com/3200x1800/?landscape,${splittedCityName.join(
              "+"
            )})`,
            backgroundSize: "cover",
          }}
        >
          <div className="search-result-card">
            <Card className="result-card-outer">
              <CardHeader
                avatar={
                  <Avatar aria-label="recipe" className={classes.avatar}>
                    {acronym}
                  </Avatar>
                }
                title={city.name}
                subheader={curTime}
              />

              <CardMedia className="city-map">
                <iframe
                  src={`https://maps.google.com/maps?q=${city.lat}, ${city.lng}&z=12&output=embed`}
                  width="100%"
                  height="300px"
                  frameBorder="0"
                ></iframe>
              </CardMedia>

              <CardContent style={{ padding: "4px" }}>
                <div className="likes">
                  <IconButton
                    className="favorite"
                    aria-label="add to favorites"
                    onClick={handleLike}
                    color="secondary"
                  >
                    {likedByMe ? <ThumbUp /> : <ThumbUpOutlined />}
                  </IconButton>
                  {likes === 0
                    ? "Ninguém curtiu essa cidade ainda"
                    : likes === 1 && !likedByMe
                    ? "Uma pessoa curtiu essa cidade"
                    : likes === 1 && likedByMe
                    ? "Apenas você curtiu essa cidade"
                    : `${likes} pessoas curtiram essa cidade`}
                </div>
              </CardContent>

              <CardContent style={{ padding: "4px" }}>
                <div className="dislikes">
                  <IconButton
                    className="favorite"
                    aria-label="add to favorites"
                    onClick={handleDislike}
                    color="secondary"
                  >
                    {dislikedByMe == true ? (
                      <ThumbDown />
                    ) : (
                      <ThumbDownAltOutlined />
                    )}
                  </IconButton>
                  {dislikes === 0
                    ? "Ninguém descurtiu essa cidade ainda"
                    : dislikes === 1 && !dislikedByMe
                    ? "Uma pessoa não curtiu essa cidade"
                    : dislikes === 1 && dislikedByMe
                    ? "Apenas você não curtiu essa cidade"
                    : `${dislikes} pessoas não curtiram essa cidade`}
                </div>
              </CardContent>

              <CardActions disableSpacing>
                <Typography variant="body2" color="textSecondary" component="p">
                  Clique para ver mais detalhes desse lugar =)
                </Typography>

                <IconButton
                  className={clsx(classes.expand, {
                    [classes.expandOpen]: expanded,
                  })}
                  onClick={handleExpandClick}
                  aria-expanded={expanded}
                  aria-label="show more"
                >
                  <ExpandMoreIcon />
                </IconButton>
              </CardActions>

              <Collapse in={expanded} timeout="auto" unmountOnExit>
                <CardContent>
                  <Typography paragraph className="country-continent">
                    <span className="label">Continente: </span>
                    Continent: {country.continent.name}
                  </Typography>

                  <Typography paragraph className="country-name">
                    <span className="label">País: </span>
                    Country: {country.name} ({country.native})
                  </Typography>

                  <Typography paragraph className="country-capital">
                    <span className="label">Capital: </span>
                    <Link to={getCityId(country.capital)}>
                      {country.capital}
                    </Link>
                  </Typography>

                  <Typography paragraph className="country-phone">
                    <span className="label">Telefone (DDI): </span>+
                    {country.phone}
                  </Typography>

                  {!!country?.currency?.length && (
                    <div className="country-currency">
                      <span className="label">
                        Moeda(s) utilizada(s) nesse país:
                      </span>
                      <ul className="currency-list">
                        {country.currency.map((cur, index) => (
                          <li key={index} className="currency-item">
                            {`1 USD equivale a ${cur.price} ${cur.unit}`}
                          </li>
                        ))}
                      </ul>
                    </div>
                  )}

                  <Typography>
                    <div className="country-languages">
                      <span className="label">
                        Linguagem(ns) utilizada(s) nesse país:
                      </span>
                      <ul className="language-list">
                        {country.languages.map((item, index) => (
                          <li key={index} className="language-item">
                            {item.name}
                          </li>
                        ))}
                      </ul>
                    </div>
                  </Typography>

                  <div className="wiki">
                    <a href={`https://pt.wikipedia.org/wiki/${city.name}`}>
                      <ReactLogo />
                    </a>
                    <a href={`https://www.dicasdeviagem.com/?s=${city.name}`}>
                      <ReactLogoMap width={100} height={100} />
                    </a>
                    <a
                      href={`https://www.instagram.com/explore/tags/${cityName
                        .split(" ")
                        .join("")}/`}
                    >
                      <ReactLogo1 className="insta_logo" />
                    </a>
                    <a
                      href={`https://www.facebook.com/search/places/?q=${city.name}`}
                    >
                      <ReactLogo2 className="insta_logo" />
                    </a>
                    <a
                      href={`https://www.google.com/search?q=${city.name}+fotos+cidade&sxsrf=ALeKk01RQe-7Q4UgGrTXi8QaHjGgEu8LTg:1606083309236&source=lnms&tbm=isch&sa=X&ved=2ahUKEwi-mYHJlpftAhV9F7kGHf0tCwUQ_AUoAnoECAUQBA&biw=1536&bih=722`}
                    >
                      <ReactLogo3 className="insta_logo" />
                    </a>
                    <a
                      href={`https://www.youtube.com/results?search_query=${city.name}+cidade`}
                    >
                      <ReactLogo4 className="insta_logo" />
                    </a>
                  </div>

                  <div className="print-share">
                    <img
                      src={qrCodeURI}
                      className="share-qrcode"
                      width="150"
                      height="150"
                    />
                    <IconButton
                      aria-label="share"
                      onClick={() => window.print()}
                    >
                      <ShareIcon />
                    </IconButton>
                    Compartilhe com seus amigos e amigas!
                  </div>
                </CardContent>
              </Collapse>
            </Card>{" "}
          </div>
        </div>
      )}
    </>
  );
}

function EmptyFeedback() {
  return <div>Não foi possível recuperar informações sobre esse lugar</div>;
}

export default SearchResult;
