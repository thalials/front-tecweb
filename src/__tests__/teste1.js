import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Testing from "../Testing";
import SearchResult from "../views/SearchResult/SearchResult";
import ResultCard from "../views/SearchResult/SearchResult";

it("should have SÃ£oPaulo in redirect", () => {
  const props = {
    match: {
      params: {
        city_id: "5fa0b349b20d4824443f6f2e",
      },
    },
  };
  const search = shallow(<ResultCard {...props} />);
  expect(toJson(search)).toMatchSnapshot();
});

it("should have nothing in redirect", () => {
  const props = {
    match: {
      params: {
        city_id: "",
      },
    },
  };
  const search = shallow(<ResultCard {...props} />);
  expect(toJson(search)).toMatchSnapshot();
});
