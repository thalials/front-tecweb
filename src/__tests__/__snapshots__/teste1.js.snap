// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should have SÃ£oPaulo in redirect 1`] = `
<div>
  <Header />
  <section
    className="search-result-main"
  >
    <ResultCard
      id="5fa0b349b20d4824443f6f2e"
    />
  </section>
</div>
`;

exports[`should have nothing in redirect 1`] = `
<div>
  <Header />
  <section
    className="search-result-main"
  >
    <ResultCard
      id=""
    />
  </section>
</div>
`;
